platform/android/export/export_plugin.cpp

```diff
@@ -416,7 +416,7 @@ void EditorExportPlatformAndroid::_check_for_changes_poll_thread(void *ud) {
 		}
 	}
 
-	if (EDITOR_GET("export/android/shutdown_adb_on_exit")) {
+	if (ea->has_runnable_preset.is_set() && EDITOR_GET("export/android/shutdown_adb_on_exit")) {
 		String adb = get_adb_path();
 		if (!FileAccess::exists(adb)) {
```

Comment:
You could add the preset check here, it would be more consistent with the check for device updates above.

Reply:
True, but there's no point running querying the editor setting or getting the adb path if we'll skip using it.

Reply:
The path won't be queried if the runnable check is false.